dashboard/config/initializers/devise.rb

```diff
@@ -286,6 +286,22 @@
   # with a log in with facebook button)
   config.omniauth :clever, CDO.dashboard_clever_key, CDO.dashboard_clever_secret, provider_ignores_state: true
 
+  config.omniauth :openid_connect, {
```

Comment:
Yeah, the [`omniauth-openid-connect`](https://github.com/jjbohn/omniauth-openid-connect) gem is a bit different from the `facebook`, `google_oauth2` etc gems because it only provides a base omniauth _strategy_ and takes additional configuration for a specific _provider_, while the latter have their endpoint configuration hard-coded into the gem. The [documentation](https://github.com/jjbohn/omniauth-openid-connect#usage) for the `name` option makes it clear that this gem supports multiple OpenID Connect providers using the same strategy:

> name is arbitrary, I recommend using the name of your provider. The name configuration exists because you could be using multiple OpenID Connect providers in a single app.

So I would distinguish the two by saying `openid_connect` is the name of the OmniAuth _strategy_, while `the_school_project` is the name of the _provider_. (Though the documentation often conflates these two concepts because they're usually identical.)

> Is the way to think about it that those other products have made their own provider that wraps the internals? And that if The School Project published a gem wrapping OpenId Connect, we could just do something like config.omniauth :the_school_project, key, key like the others? And instead of having a wrapper gem we just pass config info to openid_connect provider?

Yes, this all sounds more or less correct (though the other gems depend on the more abstract [`omniauth-oauth2`](https://github.com/intridea/omniauth-oauth2) strategy rather than `omniauth-openid-connect`).